### PR TITLE
Test validation for pushing without an image

### PR DESCRIPTION
**Pull request by alexellis**
## Description

Test validation for pushing without an image

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

I'm providing an initial test approach for @BurtonR 

## How Has This Been Tested?

Locally with `go test ./commands`

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: commands/push_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+)
+
+func Test_PushValidation(t *testing.T) {
+	testCases := []struct {
+		name     string
+		scenario string
+		image    string
+		isValid  bool
+	}{
+		{scenario: "Valid image username", name: "cli", image: "alexellis/faas-cli", isValid: true},
```
Comment (itscaro) [inline commands/push_test.go]: another one for isValid: false?

**Comment by alexellis** [comment]
Possibly. Looks like I messed up my rebase on this. I'll work on it again in the week, but I'm ok to abandon it if Burton wants to take over.

**Comment by BurtonR** [comment]
I'll take it over from here. I'll work on this in the next couple of days

File: commands/push_test.go (hunk 1 above)
Comment (alexellis) [reply commands/push_test.go]: Yes we should have more scenarios.. this is for @BurtonR to get a head-start on the tests

**Comment by alexellis** [comment]
Superseded by https://github.com/openfaas/faas-cli/pull/336

**Pull request closed without merging**